lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -838,16 +858,74 @@ ConstExprFunctionState::computeCallResult(ApplyInst *apply) {
+  switch (classifyFunction(callee)) {
   default: break;
   case WellKnownFunction::StringInitEmpty: {  // String.init()
     assert(conventions.getNumDirectSILResults() == 1 &&
            conventions.getNumIndirectSILResults() == 0 &&
            "unexpected String.init() signature");
     auto result = SymbolicValue::getString("", evaluator.getAllocator());
-    setValue(apply->getResults()[0], result);
+    setValue(apply, result);
+    return None;
+  }
+  case WellKnownFunction::StringMakeUTF8: {
+    // String.init(_builtinStringLiteral start: Builtin.RawPointer,
+    //             utf8CodeUnitCount: Builtin.Word,
+    //             isASCII: Builtin.Int1)
+    assert(conventions.getNumDirectSILResults() == 1 &&
+           conventions.getNumIndirectSILResults() == 0 &&
+           conventions.getNumParameters() == 4 &&
+           "unexpected signature");
+    auto *literal = dyn_cast<StringLiteralInst>(apply->getOperand(1));
+    if (!literal) break;
+    auto byteCount = getConstantValue(apply->getOperand(2));
+    if (byteCount.getKind() != SymbolicValue::Integer ||
+        byteCount.getIntegerValue().getLimitedValue() !=
+          literal->getValue().size())
+      break;
+    setValue(apply, SymbolicValue::getConstantInst(literal));
     return None;
   }
+  case WellKnownFunction::AllocateUninitializedArray: {
+    // We evaluate _allocateUninitialized in the flow-insensitive case because
+    // we otherwise won't find the stores to initialize the array elements.
+    // Collect them here and pretend that the array was initialized atomically.
+    if (fn)
+      break;  // Analyze flow sensitive case normally.
+
+    // This function has this signature:
+    //   func _allocateUninitializedArray<Element>(_ builtinCount: Builtin.Word)
+    //     -> (Array<Element>, Builtin.RawPointer)
+
+    // Figure out the allocation size.
+    auto numElements = getConstantValue(apply->getOperand(1));
+    if (!numElements.isConstant())
+      return numElements;
+
+    SmallVector<SILValue, 8> elements;
+    if (!ConstExprEvaluator::decodeAllocUninitializedArray(apply,
+                            numElements.getIntegerValue().getLimitedValue(),
+                                                           elements,
+                                                       /*arrayInsts*/nullptr))
+      return SymbolicValue::getUnknown((SILInstruction*)apply,
+                                       UnknownReason::Default);
+
+    // Okay, we were able to decode the array.  See if we can fold all of the
+    // elements we found.
+    SmallVector<SymbolicValue, 8> elementConstants;
+    for (auto elt : elements) {
+      auto eltCst = getConstantValue(elt);
+      if (!eltCst.isConstant())
+        return eltCst;
+      elementConstants.push_back(eltCst);
+    }
+
+    auto v = SymbolicValue::getAggregate(elementConstants,
+                                         evaluator.getAllocator());
+    //llvm::errs() << "Got it: " << v;
```

Comment:
remove this or wrap with DEBUG()?

Reply:
This is incomplete, I need to build this out when I have a way to represent arrays.  Still WIP.  I'll #if 0 is out :-)